packages/client/src/graphql/AgentChannelsGet.re

```diff
@@ -0,0 +1,149 @@
+        $channelId: String
+        $minNumberOfPredictions: Int
+        $minNumberOfQuestionsScored: Int
+     ) {
+        edges: agentChannels (
+            first: $first
+            last: $last
+            after: $after
+            before: $before
+            channelId: $channelId
+            minNumberOfPredictions: $minNumberOfPredictions
+            minNumberOfQuestionsScored: $minNumberOfQuestionsScored
+        ) {
+          total
+          pageInfo{
+            hasPreviousPage
+            hasNextPage
+            startCursor
+            endCursor
+          }
+          edges{
+            node{
+                id
+                agentId
+                channelId
+                primaryPointScore
+                numberOfPredictions
+                numberOfQuestionsScored
+                createdAt @bsDecoder(fn: "E.J.toMoment")
+                updatedAt @bsDecoder(fn: "E.J.toMoment")
+                agent {
+                    id
+                    name
+                    user: User {
+                        id
+                        name
+                        agentId
+                    }
+                    bot: Bot {
+                        id
+                        name
+                        competitorType
+                    }
+                }
+                channel {
+                    id
+                    name
+                    isArchived
+                    isPublic
+                }
+            }
+          }
+        }
+    }
+  |}
+];
+
+module QueryComponent = ReasonApollo.CreateQuery(Query);
+
+type inputType('a) =
+  (~first: int=?, ~last: int=?, ~after: string=?, ~before: string=?, unit) =>
+  'a;
+
+type direction = Primary.Connection.direction;
+
+let queryDirection = (~pageLimit, ~direction, ~fn: inputType('a), ()) =>
+  switch ((direction: direction)) {
+  | None => fn(~first=pageLimit, ())
+  | After(after) => fn(~first=pageLimit, ~after, ())
+  | Before(before) => fn(~last=pageLimit, ~before, ())
+  };
+
+let unpackResults = result =>
+  result##edges |> Rationale.Option.fmap(Primary.Connection.fromJson(toNode));
+
+let componentMaker = (query, innerComponentFn) =>
+  <QueryComponent variables=query##variables fetchPolicy="no-cache">
+    ...{o =>
+      o.result
+      |> HttpResponse.fromApollo
+      |> HttpResponse.fmap(unpackResults)
+      |> HttpResponse.optionalToMissing
+      |> innerComponentFn
+    }
+  </QueryComponent>;
+
+let component =
+    (
+      ~channelId=None,
+      ~minNumberOfPredictions=None,
+      ~minNumberOfQuestionsScored=None,
+      ~pageLimit,
+      ~direction: direction,
+      ~innerComponentFn,
+      (),
+    ) => {
+  let query =
+    queryDirection(
+      ~pageLimit,
+      ~direction,
+      ~fn=
+        Query.make(
+          ~channelId?,
+          ~minNumberOfPredictions?,
+          ~minNumberOfQuestionsScored?,
+        ),
+      (),
+    );
+  componentMaker(query, innerComponentFn);
+};
```

Comment:
Just curious, how long does it take you to write these files? They're kinda long and hypothetically we may be able to do something else to not need them to be so long later on.

Reply:
Mmm, it was fast, so there is no hard need to make scaffolding or similar at the moment.

Reply:
Cool, good work